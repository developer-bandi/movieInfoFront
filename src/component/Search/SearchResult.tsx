/*eslint-disable*/

import styled from 'styled-components';
import { Link } from 'react-router-dom';
import nullmovie from '../../tempdata/nullmovie.webp';
import { useEffect, useState } from 'react';
import { getMovieSearchResultData } from '../../modules/moviesearch';
import { css } from 'styled-components';
import { IoIosArrowBack } from 'react-icons/io';
import { IoIosArrowForward } from 'react-icons/io';

interface Info {
  name: string;
  totalpage: number;
}

interface Result {
  title: string;
  id: string;
  posterPath: string;
  rate: number;
  release: string;
}

interface SearchResultProps {
  data: {
    info: Info;
    result: Result[][];
  };
  dispatch: any;
}

const SearchResult = ({ data, dispatch }: SearchResultProps) => {
  const [start, setstart] = useState(1); //현재 페이지네이션으로 보여주고있는 첫페이지
  const [end, setend] = useState(0); //현재 페이지네이션으로 보여주고있는 마지막 페이지
  const [activepage, setactivepage] = useState(1); //현재 결과로 렌더링된 페이지

  useEffect(() => {
    window.scrollTo({
      top: 0,
    });
  });
  //리렌더링시 맨위로, 즉 페이지를 옮길때 맨위로 올라갑니다.

  useEffect(() => {
    if (data.info.totalpage <= 5) {
      setend(data.info.totalpage);
    } else {
      setend(5);
    }
  }, []);

  const MovePage = (position: number) => {
    if (data.result[activepage + position - 1] === undefined) {
      dispatch(
        getMovieSearchResultData({
          name: data.info.name,
          page: activepage + position,
        })
      );
    }
    setactivepage(activepage + position);
  };

  const prevPage = () => {
    if (activepage % 5 === 1) {
      if (end === data.info.totalpage) {
        setstart(start - 5);
        setend(end - (end % 5));
      } else {
        setstart(start - 5);
        setend(end - 5);
      }
    }
    MovePage(-1);
  };
  //다음페이지로 이동하는데, 이때 페이지네이션하는 페이지의 마지막일경우 다음 페이지네이션을 보여줍니다.
  const nextPage = () => {
    if (activepage % 5 === 0) {
      if (data.info.totalpage > end + 5) {
        setstart(start + 5);
        setend(end + 5);
      } else {
        setstart(start + 5);
        setend(data.info.totalpage);
      }
    }
    MovePage(+1);
  };
  //이전페이지로 이동하는데, 이때 페이지네이션하는 페이지의 처음일경우 이전 페이지네이션을 보여줍니다.
  return (
    <SearchResultBlock>
      {data.result[activepage - 1] === undefined
        ? null
        : data.result[activepage - 1].map((data) => {
            return (
              <PosterBlock to={`/moviedetail/${data.id}`} key={data.id}>
                <MovieImg
                  src={
                    data.posterPath === null
                      ? nullmovie
                      : `https://image.tmdb.org/t/p/w500${data.posterPath}`
                  }
                  alt="x"
                />
                <MovieInfoBlock>
                  <Title> {data.title}</Title>
                  <MovieMoreInfoBlock>
                    <InfoTitle>개봉</InfoTitle>
                    <InfoContent>
                      {data.release === '' ? '정보 없음' : data.release}
                    </InfoContent>
                  </MovieMoreInfoBlock>
                  <MovieMoreInfoBlock>
                    <InfoTitle>평점</InfoTitle>
                    <InfoContent>
                      {data.rate === 0 ? '평가중' : data.rate}
                    </InfoContent>
                  </MovieMoreInfoBlock>
                </MovieInfoBlock>
              </PosterBlock>
            );
          })}
      <PagenationBlock>
        <PagenationLeft
          onClick={prevPage}
          firstpage={activepage === 1 ? 'true' : 'false'}
        />
        {new Array(end - start + 1).fill(0).map((value, index) => {
          return (
            <PagenationButton
              active={activepage === index + start}
              onClick={function goPage(e) {
                if (data.result[index + start - 1] === undefined) {
                  dispatch(
                    getMovieSearchResultData({
                      name: data.info.name,
                      page: index + start,
                    })
                  );
                }
                setactivepage(index + start);
              }}
              key={index}
            >
              {index + start}
            </PagenationButton>
          );
        })}
        <PagenationRight
          onClick={nextPage}
          lastpage={activepage === data.info.totalpage ? 'true' : 'false'}
        />
      </PagenationBlock>
    </SearchResultBlock>
  );
};

const SearchResultBlock = styled.div`
  background: ${(props) => props.theme.background};
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 2rem;
`;

const PosterBlock = styled(Link)`
  display: flex;
  justify-content: center;
  text-decoration: none;
  color: black;
  margin-bottom: 2rem;
`;

const MovieImg = styled.img`
  width: 10rem;
  height: 15rem;
  @media screen and (max-width: 500px) {
    width: 6rem;
    height: 9rem;
  }
`;

const MovieInfoBlock = styled.div`
  width: 60rem;
  height: 13rem;
  margin-left: 3rem;
  padding: 0.5rem;
  @media screen and (max-width: 500px) {
    width: 13rem;
    height: 10rem;
    margin-left: 1.5rem;
  }
`;

const Title = styled.div`
  font-size: 25px;
  font-weight: bold;
  margin-bottom: 1rem;
  @media screen and (max-width: 500px) {
    font-size: 18px;
  }
`;

const MovieMoreInfoBlock = styled.div`
  display: flex;
  font-size: 20px;
  margin-bottom: 0.5rem;
  @media screen and (max-width: 500px) {
    font-size: 15px;
  }
`;

const InfoTitle = styled.div`
  font-weight: 550;
`;

const InfoContent = styled.div`
  margin-left: 1rem;
`;

const PagenationBlock = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 3rem;
`;
const PagenationButton = styled.div<{ active: boolean }>`
  width: 1.5rem;
  height: 1.5rem;
  border: 1px solid black;
  display: flex;
  justify-content: center;
  align-items: center;
  ${(props) =>
    props.active &&
    css`
      background: gray;
    `}
`;
const PagenationLeft = styled(IoIosArrowBack)<{ firstpage: string }>`
  display: inline;
  ${(props) =>
    props.firstpage === 'true' &&
    css`
      display: none;
    `}
`;
const PagenationRight = styled(IoIosArrowForward)<{ lastpage: string }>`
  display: inline;
  ${(props) =>
    props.lastpage === 'true' &&
    css`
      display: none;
    `}
`;
export default SearchResult;
