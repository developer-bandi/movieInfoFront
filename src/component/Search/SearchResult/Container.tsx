/*eslint-disable*/
import useSearchResult from './Hook';
import SearchResultPresentational from './Presentational';

const SearchResultContainer = () => {
  const {
    setPrevPageList,
    setNextPageList,
    selectPage,
    activepageNumber,
    searchResult,
    pageListNumber,
  } = useSearchResult();
  return (
    <SearchResultPresentational
      searchResult={searchResult}
      setPrevPageList={setPrevPageList}
      setNextPageList={setNextPageList}
      activepageNumber={activepageNumber}
      selectPage={selectPage}
      pageListNumber={pageListNumber}
    />
  );
};

export default SearchResultContainer;
