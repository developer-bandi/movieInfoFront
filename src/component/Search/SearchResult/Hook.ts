/*eslint-disable*/
import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { ReducerType } from '../../../store';
import {
  initializeMovieSearch,
  searchMovie,
} from '../../../store/movieSearch/Reducer';

const useSearchResult = () => {
  const [pageListNumber, setPageListNumber] = useState(1);
  const [activepageNumber, setactivepageNumber] = useState(1); //현재 결과로 렌더링된 페이지
  const searchResult = useSelector((state: ReducerType) => state.movieSearch);
  const dispatch = useDispatch();

  useEffect(() => {
    window.scrollTo({
      top: 0,
    });
  });
  //리렌더링시 맨위로, 즉 페이지를 옮길때 맨위로 올라갑니다.

  useEffect(() => {
    if (searchResult.content.searchResult?.page === 1) {
      setPageListNumber(1);
      setactivepageNumber(1);
    }
  }, [searchResult.content.keyword, searchResult.content.searchResult?.page]);

  useEffect(() => {
    return () => {
      dispatch(initializeMovieSearch());
    };
  }, []);

  const MovePage = (position: number) => {
    dispatch(
      searchMovie({
        keyword: searchResult.content.keyword,
        page: position,
      })
    );
    setactivepageNumber(position);
  };

  const setPrevPageList = () => {
    if (searchResult.content.searchResult !== undefined) {
      if (pageListNumber !== 1) {
        setPageListNumber(pageListNumber - 1);
        MovePage((pageListNumber - 2) * 5 + 1);
      }
    }
  };
  //다음페이지로 이동하는데, 이때 페이지네이션하는 페이지의 마지막일경우 다음 페이지네이션을 보여줍니다.
  const setNextPageList = () => {
    if (searchResult.content.searchResult !== undefined) {
      if (searchResult.content.searchResult.totalPage > pageListNumber * 5) {
        setPageListNumber(pageListNumber + 1);
        MovePage(pageListNumber * 5 + 1);
      }
    }
  };
  //이전페이지로 이동하는데, 이때 페이지네이션하는 페이지의 처음일경우 이전 페이지네이션을 보여줍니다.

  const selectPage = (index: number) => {
    dispatch(
      searchMovie({
        keyword: searchResult.content.keyword,
        page: index + (pageListNumber - 1) * 5,
      })
    );
    setactivepageNumber(index + (pageListNumber - 1) * 5);
  };
  return {
    setPrevPageList,
    setNextPageList,
    selectPage,
    activepageNumber,
    searchResult,
    pageListNumber,
  };
};

export default useSearchResult;
