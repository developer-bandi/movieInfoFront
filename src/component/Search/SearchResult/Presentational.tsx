/*eslint-disable*/
import nullmovie from '../../../imgs/nullmovie.webp';
import { IoIosArrowBack } from 'react-icons/io';
import { IoIosArrowForward } from 'react-icons/io';
import styles from './Style';
import { MovieSearchState } from '../../../store/movieSearch/Reducer';
import Loading from '../../common/Loading/Presentational';
import Except from '../../common/Except/Presentational';

interface SearchResultProps {
  searchResult: MovieSearchState;
  setPrevPageList: () => void;
  setNextPageList: () => void;
  activepageNumber: number;
  selectPage: (index: number) => void;
  pageListNumber: number;
}

const SearchResultPresentational = ({
  searchResult,
  setPrevPageList,
  setNextPageList,
  activepageNumber,
  selectPage,
  pageListNumber,
}: SearchResultProps) => {
  if (searchResult.loading) {
    return <Loading data-testid={'loading'}></Loading>;
  } else if (searchResult.error) {
    return <Except text="에러가 발생하였습니다" data-testid={'error'} />;
  } else if (searchResult.content.searchResult === undefined) {
    return <Except text="" data-testid={'blank'} />;
  } else if (searchResult.content.searchResult.results.length === 0) {
    return <Except text="검색 결과가 없습니다" data-testid={'noResult'} />;
  } else {
    return (
      <styles.MainBlock>
        {searchResult.content.searchResult.results.map((poserInfo) => {
          return (
            <styles.PosterBlock
              to={`/moviedetail/${poserInfo.id}`}
              key={poserInfo.id}
            >
              <styles.MovieImg
                src={
                  poserInfo.posterPath === null
                    ? nullmovie
                    : `https://image.tmdb.org/t/p/w500${poserInfo.posterPath}`
                }
                alt={poserInfo.posterPath === null ? 'notPoster' : 'poster'}
              />
              <styles.MovieInfoBlock>
                <styles.Title> {poserInfo.title}</styles.Title>
                <styles.MovieMoreInfoBlock>
                  <styles.InfoTitle>개봉</styles.InfoTitle>
                  <styles.InfoContent>{poserInfo.release}</styles.InfoContent>
                </styles.MovieMoreInfoBlock>
                <styles.MovieMoreInfoBlock>
                  <styles.InfoTitle>평점</styles.InfoTitle>
                  <styles.InfoContent>
                    {poserInfo.rate === 0 ? '평가중' : poserInfo.rate}
                  </styles.InfoContent>
                </styles.MovieMoreInfoBlock>
              </styles.MovieInfoBlock>
            </styles.PosterBlock>
          );
        })}
        <styles.PagenationBlock>
          {pageListNumber !== 1 ? (
            <styles.MovePage onClick={setPrevPageList}>
              <IoIosArrowBack data-testid={'prev'} size={20} />
            </styles.MovePage>
          ) : null}
          {new Array(
            pageListNumber * 5 >= searchResult.content.searchResult.totalPage
              ? searchResult.content.searchResult.totalPage -
                (pageListNumber - 1) * 5
              : 5
          )
            .fill(0)
            .map((value, index) => {
              return (
                <styles.PageButton
                  active={
                    activepageNumber === index + 1 + (pageListNumber - 1) * 5
                  }
                  onClick={() => selectPage(index + 1)}
                  key={index}
                >
                  {index + 1 + (pageListNumber - 1) * 5}
                </styles.PageButton>
              );
            })}
          {pageListNumber * 5 < searchResult.content.searchResult.totalPage ? (
            <styles.MovePage onClick={setNextPageList}>
              <IoIosArrowForward data-testid={'next'} size={20} />
            </styles.MovePage>
          ) : null}
        </styles.PagenationBlock>
      </styles.MainBlock>
    );
  }
};

export default SearchResultPresentational;
