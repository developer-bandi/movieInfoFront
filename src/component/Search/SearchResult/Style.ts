import { css } from 'styled-components';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

const MainBlock = styled.div`
  background: ${(props) => props.theme.background};
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-top: 2rem;
`;

const PosterBlock = styled(Link)`
  display: flex;
  justify-content: center;
  text-decoration: none;
  color: black;
  margin-bottom: 2rem;
`;

const MovieImg = styled.img`
  width: 10rem;
  height: 15rem;
  @media screen and (max-width: 500px) {
    width: 6rem;
    height: 9rem;
  }
`;

const MovieInfoBlock = styled.div`
  width: 60rem;
  height: 13rem;
  margin-left: 3rem;
  padding: 0.5rem;
  @media screen and (max-width: 500px) {
    width: 13rem;
    height: 10rem;
    margin-left: 1.5rem;
  }
`;

const Title = styled.div`
  font-size: 25px;
  font-weight: bold;
  margin-bottom: 1rem;
  @media screen and (max-width: 500px) {
    font-size: 18px;
  }
`;

const MovieMoreInfoBlock = styled.div`
  display: flex;
  font-size: 20px;
  margin-bottom: 0.5rem;
  @media screen and (max-width: 500px) {
    font-size: 15px;
  }
`;

const InfoTitle = styled.div`
  font-weight: 550;
`;

const InfoContent = styled.div`
  margin-left: 1rem;
`;

const PagenationBlock = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 3rem;
`;
const PageButton = styled.div<{ active: boolean }>`
  width: 1.5rem;
  height: 1.5rem;
  border: 1px solid black;
  display: flex;
  justify-content: center;
  align-items: center;
  ${(props) =>
    props.active &&
    css`
      background: gray;
    `}
`;

const MovePage = styled.div<{ active: string }>`
  display: inline;
  ${(props) =>
    props.active === 'true' &&
    css`
      display: none;
    `}
`;

const styles = {
  MainBlock,
  PosterBlock,
  MovieImg,
  MovieInfoBlock,
  Title,
  MovieMoreInfoBlock,
  InfoTitle,
  InfoContent,
  PagenationBlock,
  PageButton,
  MovePage,
};

export default styles;
