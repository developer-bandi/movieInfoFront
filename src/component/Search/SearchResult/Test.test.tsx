import { render, screen } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import useSearchResult from './Hook';
import SearchResult from './Presentational';
const state = {
  content: {
    searchResult: {
      page: 1,
      totalPage: 15,
      results: [
        {
          title: 'test',
          id: 123456,
          posterPath: 'testPath',
          rate: 9.0,
          release: 'testDate',
        },
      ],
    },
    keyword: 'test',
  },
  loading: false,
  error: false,
};
describe('SearchResult Presentational', () => {
  it('로딩중', () => {
    const utils = render(
      <SearchResult
        searchResult={{
          content: { keyword: '' },
          loading: true,
          error: false,
        }}
        setPrevPageList={jest.fn()}
        setNextPageList={jest.fn()}
        activepageNumber={1}
        selectPage={jest.fn()}
        pageListNumber={1}
      />
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId('loading');
  });
  it('에러 발생', () => {
    const utils = render(
      <SearchResult
        searchResult={{
          content: { keyword: '' },
          loading: false,
          error: true,
        }}
        setPrevPageList={jest.fn()}
        setNextPageList={jest.fn()}
        activepageNumber={1}
        selectPage={jest.fn()}
        pageListNumber={1}
      />
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId('except');
  });
  it('검색 이전', () => {
    const utils = render(
      <SearchResult
        searchResult={{
          content: {
            searchResult: {
              page: 1,
              totalPage: 1,
              results: [],
            },
            keyword: '',
          },
          loading: false,
          error: true,
        }}
        setPrevPageList={jest.fn()}
        setNextPageList={jest.fn()}
        activepageNumber={1}
        selectPage={jest.fn()}
        pageListNumber={1}
      />
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId('blank');
  });
  it('결과 없음', () => {
    const utils = render(
      <SearchResult
        searchResult={{
          content: {
            searchResult: {
              page: 1,
              totalPage: 1,
              results: [],
            },
            keyword: '',
          },
          loading: false,
          error: true,
        }}
        setPrevPageList={jest.fn()}
        setNextPageList={jest.fn()}
        activepageNumber={1}
        selectPage={jest.fn()}
        pageListNumber={1}
      />
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId('except');
  });
  describe('정상 출력', () => {
    it('검색 리스트 출력 검증', () => {
      const utils = render(
        <SearchResult
          searchResult={state}
          setPrevPageList={jest.fn()}
          setNextPageList={jest.fn()}
          activepageNumber={1}
          selectPage={jest.fn()}
          pageListNumber={1}
        />,
        { wrapper: MemoryRouter }
      );
      expect(utils.container).toMatchSnapshot();
    });
    describe('페이지 버튼 출력 검증', () => {
      it('첫 페이지 리스트', () => {
        const utils = render(
          <SearchResult
            searchResult={state}
            setPrevPageList={jest.fn()}
            setNextPageList={jest.fn()}
            activepageNumber={1}
            selectPage={jest.fn()}
            pageListNumber={1}
          />,
          { wrapper: MemoryRouter }
        );
        expect(utils.container).toMatchSnapshot();
        screen.getByTestId('next');
      });
      it('중간 리스트', () => {
        const utils = render(
          <SearchResult
            searchResult={state}
            setPrevPageList={jest.fn()}
            setNextPageList={jest.fn()}
            activepageNumber={1}
            selectPage={jest.fn()}
            pageListNumber={2}
          />,
          { wrapper: MemoryRouter }
        );
        expect(utils.container).toMatchSnapshot();
        screen.getByTestId('prev');
        screen.getByTestId('next');
      });
      it('마지막 리스트', () => {
        const utils = render(
          <SearchResult
            searchResult={state}
            setPrevPageList={jest.fn()}
            setNextPageList={jest.fn()}
            activepageNumber={1}
            selectPage={jest.fn()}
            pageListNumber={3}
          />,
          { wrapper: MemoryRouter }
        );
        expect(utils.container).toMatchSnapshot();
        screen.getByTestId('prev');
      });
    });
  });
});

describe('SearchResult Hook', () => {
  window.scrollTo = jest.fn();
  const mockStore = configureMockStore();
  const store = mockStore({ movieSearch: state });
  const wrapper = ({ children }: { children: React.FC }) => (
    <Provider store={store}>{children}</Provider>
  );
  it('setNextPageList,setPrevPageList 함수 작동 여부', () => {
    const { result } = renderHook(() => useSearchResult(), {
      wrapper,
    });
    act(() => {
      result.current.setNextPageList();
    });
    expect(store.getActions()[0]).toStrictEqual({
      type: 'movieSearch/searchMovie',
      payload: { keyword: 'test', page: 6 },
    });
    expect(result.current.pageListNumber).toBe(2);
    expect(result.current.activepageNumber).toBe(6);
    act(() => {
      result.current.setPrevPageList();
    });
    expect(store.getActions()[1]).toStrictEqual({
      type: 'movieSearch/searchMovie',
      payload: { keyword: 'test', page: 1 },
    });
    expect(result.current.pageListNumber).toBe(1);
    expect(result.current.activepageNumber).toBe(1);
  });
  it('selectPage 함수 작동 여부', () => {
    const { result } = renderHook(() => useSearchResult(), {
      wrapper,
    });
    act(() => {
      result.current.selectPage(2);
    });
    expect(store.getActions()[3]).toStrictEqual({
      type: 'movieSearch/searchMovie',
      payload: { keyword: 'test', page: 2 },
    });
    expect(result.current.activepageNumber).toBe(2);
  });
});
