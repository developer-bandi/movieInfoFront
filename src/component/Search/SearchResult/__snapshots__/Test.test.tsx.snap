// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult Presentational 결과 없음 1`] = `
<div>
  <div
    class="sc-crHmcD hAuXPM"
  >
    <div
      class="sc-egiyK fcOVsL"
      data-testid="except"
    >
      에러가 발생하였습니다
    </div>
  </div>
</div>
`;

exports[`SearchResult Presentational 로딩중 1`] = `
<div>
  <div
    class="sc-kDTinF hjqJdy"
  >
    <img
      class="sc-iqseJM dwFJvL"
      data-testid="loading"
      src="loading.gif"
    />
    ;
  </div>
</div>
`;

exports[`SearchResult Presentational 에러 발생 1`] = `
<div>
  <div
    class="sc-crHmcD hAuXPM"
  >
    <div
      class="sc-egiyK fcOVsL"
      data-testid="except"
    >
      에러가 발생하였습니다
    </div>
  </div>
</div>
`;

exports[`SearchResult Presentational 정상 출력 검색 리스트 출력 검증 1`] = `
<div>
  <div
    class="sc-bdvvtL ckqPWU"
  >
    <a
      class="sc-gsDKAQ YHYaT"
      href="/moviedetail/123456"
    >
      <img
        alt="poster"
        class="sc-dkPtRN eZiIAo"
        src="https://image.tmdb.org/t/p/w500testPath"
      />
      <div
        class="sc-hKwDye gyUwbE"
      >
        <div
          class="sc-eCImPb fJhZTf"
        >
           
          test
        </div>
        <div
          class="sc-jRQBWg hHgpWM"
        >
          <div
            class="sc-gKclnd lgENuB"
          >
            개봉
          </div>
          <div
            class="sc-iCfMLu jLUtnt"
          >
            testDate
          </div>
        </div>
        <div
          class="sc-jRQBWg hHgpWM"
        >
          <div
            class="sc-gKclnd lgENuB"
          >
            평점
          </div>
          <div
            class="sc-iCfMLu jLUtnt"
          >
            9
          </div>
        </div>
      </div>
    </a>
    <div
      class="sc-furwcr hRrBwF"
    >
      <div
        class="sc-pVTFL cVaegz"
      >
        0
      </div>
      <div
        class="sc-pVTFL jBzWTV"
      >
        1
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        2
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        3
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        4
      </div>
      <div
        class="sc-jrQzAO iLjvwk"
      >
        <svg
          data-testid="next"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchResult Presentational 정상 출력 페이지 버튼 출력 검증 마지막 리스트 1`] = `
<div>
  <div
    class="sc-bdvvtL ckqPWU"
  >
    <a
      class="sc-gsDKAQ YHYaT"
      href="/moviedetail/123456"
    >
      <img
        alt="poster"
        class="sc-dkPtRN eZiIAo"
        src="https://image.tmdb.org/t/p/w500testPath"
      />
      <div
        class="sc-hKwDye gyUwbE"
      >
        <div
          class="sc-eCImPb fJhZTf"
        >
           
          test
        </div>
        <div
          class="sc-jRQBWg hHgpWM"
        >
          <div
            class="sc-gKclnd lgENuB"
          >
            개봉
          </div>
          <div
            class="sc-iCfMLu jLUtnt"
          >
            testDate
          </div>
        </div>
        <div
          class="sc-jRQBWg hHgpWM"
        >
          <div
            class="sc-gKclnd lgENuB"
          >
            평점
          </div>
          <div
            class="sc-iCfMLu jLUtnt"
          >
            9
          </div>
        </div>
      </div>
    </a>
    <div
      class="sc-furwcr hRrBwF"
    >
      <div
        class="sc-jrQzAO iLjvwk"
      >
        <svg
          data-testid="prev"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        10
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        11
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        12
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        13
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        14
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchResult Presentational 정상 출력 페이지 버튼 출력 검증 중간 리스트 1`] = `
<div>
  <div
    class="sc-bdvvtL ckqPWU"
  >
    <a
      class="sc-gsDKAQ YHYaT"
      href="/moviedetail/123456"
    >
      <img
        alt="poster"
        class="sc-dkPtRN eZiIAo"
        src="https://image.tmdb.org/t/p/w500testPath"
      />
      <div
        class="sc-hKwDye gyUwbE"
      >
        <div
          class="sc-eCImPb fJhZTf"
        >
           
          test
        </div>
        <div
          class="sc-jRQBWg hHgpWM"
        >
          <div
            class="sc-gKclnd lgENuB"
          >
            개봉
          </div>
          <div
            class="sc-iCfMLu jLUtnt"
          >
            testDate
          </div>
        </div>
        <div
          class="sc-jRQBWg hHgpWM"
        >
          <div
            class="sc-gKclnd lgENuB"
          >
            평점
          </div>
          <div
            class="sc-iCfMLu jLUtnt"
          >
            9
          </div>
        </div>
      </div>
    </a>
    <div
      class="sc-furwcr hRrBwF"
    >
      <div
        class="sc-jrQzAO iLjvwk"
      >
        <svg
          data-testid="prev"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        5
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        6
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        7
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        8
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        9
      </div>
      <div
        class="sc-jrQzAO iLjvwk"
      >
        <svg
          data-testid="next"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchResult Presentational 정상 출력 페이지 버튼 출력 검증 첫 페이지 리스트 1`] = `
<div>
  <div
    class="sc-bdvvtL ckqPWU"
  >
    <a
      class="sc-gsDKAQ YHYaT"
      href="/moviedetail/123456"
    >
      <img
        alt="poster"
        class="sc-dkPtRN eZiIAo"
        src="https://image.tmdb.org/t/p/w500testPath"
      />
      <div
        class="sc-hKwDye gyUwbE"
      >
        <div
          class="sc-eCImPb fJhZTf"
        >
           
          test
        </div>
        <div
          class="sc-jRQBWg hHgpWM"
        >
          <div
            class="sc-gKclnd lgENuB"
          >
            개봉
          </div>
          <div
            class="sc-iCfMLu jLUtnt"
          >
            testDate
          </div>
        </div>
        <div
          class="sc-jRQBWg hHgpWM"
        >
          <div
            class="sc-gKclnd lgENuB"
          >
            평점
          </div>
          <div
            class="sc-iCfMLu jLUtnt"
          >
            9
          </div>
        </div>
      </div>
    </a>
    <div
      class="sc-furwcr hRrBwF"
    >
      <div
        class="sc-pVTFL cVaegz"
      >
        0
      </div>
      <div
        class="sc-pVTFL jBzWTV"
      >
        1
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        2
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        3
      </div>
      <div
        class="sc-pVTFL cVaegz"
      >
        4
      </div>
      <div
        class="sc-jrQzAO iLjvwk"
      >
        <svg
          data-testid="next"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M294.1 256L167 129c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.3 34 0L345 239c9.1 9.1 9.3 23.7.7 33.1L201.1 417c-4.7 4.7-10.9 7-17 7s-12.3-2.3-17-7c-9.4-9.4-9.4-24.6 0-33.9l127-127.1z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
